packages/mui-base/src/useSnackbar/useSnackbar.ts

```diff
@@ -145,9 +145,11 @@ export default function useSnackbar(parameters: UseSnackbarParameters) {
     return undefined;
   }, [disableWindowBlurListener, handleResume, open]);
 
-  const getRootProps = <TOther extends Record<string, React.EventHandler<any> | undefined> = {}>(
-    otherHandlers: TOther = {} as TOther,
-  ): UseSnackbarRootSlotProps<TOther> => {
+  const getRootProps: UseSnackbarReturnValue['getRootProps'] = <
+    TOther extends Parameters<UseSnackbarReturnValue['getRootProps']>[0],
+  >(
+    otherHandlers: TOther,
```

Comment:
We should add default value as well, in case an event handler is not passed.
```suggestion
    otherHandlers: TOther = {} as TOther,
```